test/e2e/provider/bfcache.spec.js

```diff
@@ -0,0 +1,72 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  defaultGanacheOptions,
+  DAPP_URL,
+  openDapp,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+const triggerBFCache = async (driver) => {
+  await driver.executeScript(`
+    window.addEventListener('pageshow', (event) => {
+      if (event.persisted) {
+        window.restoredFromBFCache = true
+      }
+    });
+  `);
+
+  let attempts = 0;
+  while (attempts < 10) {
```

Comment:
why do we need multiple attempts, and why `10` and not some other number?

Reply:
A page doesn't get BFCached for me consistently. It usually works on the first attempt though. What if we bring this down to 3?

Reply:
looks like i'm wrong and we don't need this anymore. Removed loop here https://github.com/MetaMask/metamask-extension/pull/24950/commits/2dfb287d59490312c700cd4f45f5bf57040462d8